tests/Doctrine/Tests/DBAL/Functional/Schema/SchemaManagerFunctionalTestCase.php

```diff
@@ -42,6 +43,29 @@ protected function setUp()
         $this->_sm = $this->_conn->getSchemaManager();
     }
 
+
+    protected function tearDown()
+    {
+        parent::tearDown();
+
+        if ($this->getName() !== 'testTableInNamespace' || ! $this->_sm->getDatabasePlatform()->supportsSchemas()) {
```

Comment:
As far as I understand, this is in order to prevent the exception from `dropTable()` if the table doesn't exist. This is not the best solution since the table may not exist if the test failed and didn't manage to create it.

Please use `$this->_sm->tryMethod()` and get rid of the conditional logic.

Reply:
Actually, this is done to reduce the attempts to drop the table / the namespace.
However, I think performance is no requirement on functional tests. So, I will remove it.
But there will be a special handling for the `DROP SCHEMA` left.